### ISSUE TITLE
Error outputs for dynamic height and width

### ISSUE THREAD

**Issue by col-in-coding** [NVIDIA/TensorRT#2457]
## Description

I'm doing some tensorrt deployment works for dynamic height and width input, but only the outputs with max width input were correct. 

## Environment

**TensorRT Version**: 8.4.3.1
**NVIDIA GPU**: 3090
**NVIDIA Driver Version**: 510.47.03
**CUDA Version**: 11.6
**CUDNN Version**: 8.4
**Operating System**: Ubuntu18.04
**Python Version (if applicable)**: 3.7.9
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 1.12.1
**Baremetal or Container (if so, version)**: 

## Relevant Files

To reproduce this issue, I created a demo script here
https://github.com/col-in-coding/Tensorrt-CV/blob/master/dynamic_shape/dynamic_shape_problem.py

## Steps To Reproduce

```
dynamic_shape_problem.py --build
dynamic_shape_problem.py --run
```

https://github.com/NVIDIA/TensorRT/issues/2457#tasklist-block-19dc098d-ce43-415c-9d25-2f4ee86cf6f2

**Comment by zerollzeng** [comment]
I believe you need to set input shapes if dynamic shapes is enabled. but I'm not an expert in the python api. @pranavm-nvidia do you have any suggestion here?

**Comment by col-in-coding** [comment]
> u need to set input sha

You mean set binding set in context? I did it

**Comment by col-in-coding** [comment]
> I believe you need to set input shapes if dynamic shapes is enabled. but I'm not an expert in the python api. @pranavm-nvidia do you have any suggestion here?

You mean set binding set in context? I did it

**Comment by col-in-coding** [comment]
I find its tricky, my problem could be solved by multi-profiling it. But I have no idea why the single profile incorrect.

**Comment by col-in-coding** [comment]
> That's weird. You could try using the Polygraphy Python API to rule out a bug in your code. A couple examples that might help:
> 
> * https://github.com/NVIDIA/TensorRT/tree/main/tools/Polygraphy/examples/api/00_inference_with_tensorrt
> * https://github.com/NVIDIA/TensorRT/tree/main/tools/Polygraphy/examples/api/07_tensorrt_and_dynamic_shapes

I run this from cli:
```
polygraphy run dytest.onnx --trt --onnxrt \
--onnx-outputs mark all \
--trt-outputs mark all \
--trt-min-shapes input:[1,2,32,32] --trt-opt-shapes input:[1,2,64,64] --trt-max-shapes input:[1,2,64,64] \
--input-shapes input:[1,2,32,64] \
--atol 1e-5 --rtol 1e-5 \
--pool-limit workspace:5G
``` 
But it gives me an error
```
[I] Will generate inference input data according to provided TensorMetadata: {input [shape=(1, 2, 32, 64)]}
[I] trt-runner-N0-11/10/22-12:11:17     | Activating and starting inference
[W] onnx2trt_utils.cpp:369: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[I]     Configuring with profiles: [Profile().add('input', min=[1, 2, 32, 32], opt=[1, 2, 64, 64], max=[1, 2, 64, 64])]
[W] It looks like some layers in the network have compute precision set, but precision constraints were not enabled. 
    Precision constraints must be set to 'prefer' or 'obey' for layer compute precision to take effect. 
    Note: Layers and their requested precisions were: {'Shape_3': 'INT32', 'Shape_6': 'INT32', 'Shape_9': 'INT32', 'Shape_12': 'INT32', 'Shape_25': 'INT32', 'Shape_28': 'INT32', 'Shape_31': 'INT32', 'Shape_47': 'INT32', 'Shape_52': 'INT32', 'Shape_55': 'INT32', 'Shape_70': 'INT32', 'Shape_73': 'INT32', 'Shape_76': 'INT32', 'Shape_92': 'INT32', 'Shape_97': 'INT32', 'Shape_100': 'INT32', 'Shape_115': 'INT32', 'Shape_118': 'INT32', 'Shape_121': 'INT32', 'Shape_137': 'INT32', 'Shape_142': 'INT32', 'Shape_145': 'INT32', '(Unnamed Layer* 155) [Shape]': 'INT32', 'Shape_166': 'INT32', 'Shape_173': 'INT32', 'Shape_176': 'INT32', 'Shape_188': 'INT32', 'Shape_195': 'INT32', 'Shape_198': 'INT32'}
[E] 3: [builderConfig.cpp::getFlag::42] Error Code 3: API Usage Error (Parameter check failed at: optimizer/api/builderConfig.cpp::getFlag::42, condition: int(builderFlag) >= 0 && int(builderFlag) < EnumMax<BuilderFlag>()
    )
poly_run.sh: line 7: 44983 Segmentation fault      (core dumped) polygraphy run dytest.onnx --trt --onnxrt --onnx-outputs mark all --trt-outputs mark all --trt-min-shapes input:[1,2,32,32] --trt-opt-shapes input:[1,2,64,64] --trt-max-shapes input:[1,2,64,64] --input-shapes input:[1,2,32,64] --atol 1e-5 --rtol 1e-5 --pool-limit workspace:5G
```

**Comment by col-in-coding** [comment]
> Ah that usually happens when your TRT library version and python bindings version is mismatched. How did you install TRT?
Yes, now I got this error.
···
[W] Skipping tactic 0x0000000000000000 due to Myelin error: Formal output tensor "onnx__MatMul_242 _ (Unnamed Layer_ 22) [Shuffle]_constant" is also a data tensor.
[E] 10: [optimizer.cpp::computeCosts::3626] Error Code 10: Internal Error (Could not find any implementation for node {ForeignNode[onnx::MatMul_242 + (Unnamed Layer* 22) [Shuffle]...Reshape_214]}.)
[E] 2: [builder.cpp::buildSerializedNetwork::636] Error Code 2: Internal Error (Assertion engine != nullptr failed. )
[!] Invalid Engine. Please ensure the engine was built correctly
···

**Comment by col-in-coding** [comment]
![image](https://user-images.githubusercontent.com/15387667/201245209-8d605b1f-31f3-4e9f-aac4-8d33829fc959.png)

**Comment by col-in-coding** [comment]
@pranavm-nvidia @zerollzeng 
Hey guys, the Polygraphy tells me something wrong with the Transpose layer, but I have no idea what is wrong.

https://github.com/col-in-coding/Tensorrt-CV/blob/master/dynamic_shape/poly_run.sh
https://github.com/col-in-coding/Tensorrt-CV/blob/master/dynamic_shape/log.log

**Comment by pranavm-nvidia** [comment]
Seems likely to be a TRT bug. Does this happen even with 8.5?

**Comment by col-in-coding** [comment]
> Seems likely to be a TRT bug. Does this happen even with 8.5?

Yes, I'm using V8.5

**Comment by col-in-coding** [comment]
I find this no happen in V7.2.3

**Comment by syyxsxx** [comment]
> > Seems likely to be a TRT bug. Does this happen even with 8.5?
> 
> Yes, I'm using V8.5

the bug can be slove by set tensorrt.PreviewFeature when convert model

**Comment by col-in-coding** [comment]
> > > Seems likely to be a TRT bug. Does this happen even with 8.5?
> > 
> > 
> > Yes, I'm using V8.5
> 
> the bug can be slove by set tensorrt.PreviewFeature when convert model

Bravo!

**Comment by col-in-coding** [comment]
I have added the solution for this issue
https://github.com/col-in-coding/Tensorrt-CV/commit/eb99520b9c3209ded4ef0ac79533395ba3a222dc

**Comment by henbucuoshanghai** [comment]
how about  the dynamic batch input? where is the demo?
